### PR TITLE
StalinSort.java

### PR DESCRIPTION
**Pull request by anant-jain01**
- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.
- [x] All new code is formatted with `clang-format -i --style=file path/to/your/file.java`

### CODE REVIEW

**Comment by anant-jain01** [comment]
@yanglbme @vil02 @BamaCharanChhandogi @alxkm @siriak 
please check and if you choose to merge
please use the hacktoberfest accepted tag

File: src/main/java/com/thealgorithms/sorts/AdaptiveMergeSort.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+                array[k] = aux[i++];
+            } else if (aux[j].compareTo(aux[i]) < 0) {
+                array[k] = aux[j++];
+            } else {
+                array[k] = aux[i++];
+            }
+        }
+    }
+
+    public static void main(String[] args) {
```
Comment (siriak) [inline src/main/java/com/thealgorithms/sorts/AdaptiveMergeSort.java:43]: Please remove main

File: src/main/java/com/thealgorithms/sorts/StalinSort.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        }
+
+        // Create a result array with sorted elements
+        T[] result = (T[]) java.lang.reflect.Array.newInstance(array.getClass().getComponentType(), currentIndex + 1);
+        System.arraycopy(array, 0, result, 0, currentIndex + 1);
+        return result;
+    }
+
+    // Driver Program
+    public static void main(String[] args) {
```
Comment (siriak) [inline src/main/java/com/thealgorithms/sorts/StalinSort.java:25]: And here

**Comment by anant-jain01** [comment]
@siriak the changes have ben made.
Please check it out...

**Review by siriak — APPROVED** [review]
Looks good, thanks!

**Pull request merged**